File: funcscript/funcs/list/TakeFunction.cs
using funcscript.core;
using funcscript.model;
using System;

namespace funcscript.funcs.list
{
    public class TakeFunction : IFsFunction
    {
        public int MaxParsCount => 2;

        public CallType CallType => CallType.Prefix;

        public string Symbol => "Take";

        public int Precidence => 0;

        public object Evaluate(IFsDataProvider parent, IParameterList pars)
        {
            if (pars.Count != this.MaxParsCount)
                throw new error.TypeMismatchError($"{this.Symbol} function: Invalid parameter count. Expected {this.MaxParsCount}, but got {pars.Count}");

            var par0 = pars.GetParameter(parent,0);
            var par1 = pars.GetParameter(parent,1);

            return EvaluateInternal(par0, par1);
        }

        private object EvaluateInternal(object par0, object par1)
        {
            if (par0 == null)
                return null;

            if (!(par0 is FsList))
                throw new error.TypeMismatchError($"{this.Symbol} function: The first parameter should be {this.ParName(0)}");

            if (!(par1 is int))
                throw new error.TypeMismatchError($"{this.Symbol} function: The second parameter should be {this.ParName(1)}");

            var lst = (FsList)par0;
            int n = (int)par1;

            if (n <= 0)
                return new ArrayFsList(new object[] { });

            if (n > lst.Length)
                n = lst.Length;

            return new ArrayFsList(lst.Take(n).ToArray());
        }


        public string ParName(int index)
        {
            return index switch
            {
                 0 => "List",
                 1 => "Number",
                _ => ""
            };
        }
    }
}

File: js-port/funscscript-js/src/funcs/list/take-function.js
const { BaseFunction, CallType } = require('../../core/function-base');
const { ArrayFsList } = require('../../model/fs-list');
const helpers = require('../helpers');
const { FSDataType } = require('../../core/fstypes');

class TakeFunction extends BaseFunction {
  constructor() {
    super();
    this.symbol = 'Take';
    this.callType = CallType.Prefix;
  }

  get maxParameters() {
    return 2;
  }

  evaluate(provider, parameters) {
    const error = helpers.expectParamCount(this.symbol, parameters, this.maxParameters);
    if (error) {
      return error;
    }

    const list = helpers.ensureList(parameters.getParameter(provider, 0));
    if (!list) {
      return helpers.makeError(
        helpers.FsError.ERROR_TYPE_MISMATCH,
        `${this.symbol} function: The first parameter should be ${this.parName(0)}`
      );
    }

    const countResult = helpers.requireInteger(this.symbol, parameters.getParameter(provider, 1), 'count');
    if (!countResult.ok) {
      return countResult.error;
    }
    let count = countResult.value;
    if (count <= 0) {
      return helpers.makeValue(FSDataType.List, new ArrayFsList([]));
    }

    if (count > list.length) {
      count = list.length;
    }

    const values = [];
    for (let i = 0; i < count; i += 1) {
      values.push(list.get(i));
    }
    return helpers.makeValue(FSDataType.List, new ArrayFsList(values));
  }

  parName(index) {
    return index === 0 ? 'List' : index === 1 ? 'Number' : '';
  }
}

module.exports = {
  TakeFunction
};
